src/libprojectM/CMakeLists.txt

```diff
@@ -0,0 +1,245 @@
+        ConfigFile.h
+        dlldefs.h
+        event.h
+        fatal.h
+        fftsg.cpp
+        fftsg.h
+        FileScanner.cpp
+        FileScanner.hpp
+        glError.h
+        gltext.h
+        HungarianMethod.hpp
+        IdleTextures.hpp
+        KeyHandler.cpp
+        KeyHandler.hpp
+        lvtoprojectM.h
+        PCM.cpp
+        PCM.hpp
+        PipelineMerger.cpp
+        PipelineMerger.hpp
+        Preset.cpp
+        Preset.hpp
+        PresetChooser.cpp
+        PresetChooser.hpp
+        PresetFactory.cpp
+        PresetFactory.hpp
+        PresetFactoryManager.cpp
+        PresetFactoryManager.hpp
+        PresetLoader.cpp
+        PresetLoader.hpp
+        projectM.cpp
+        projectM.hpp
+        projectM-opengl.h
+        RandomNumberGenerators.hpp
+        resource.h
+        sdltoprojectM.h
+        TestRunner.cpp
+        TestRunner.hpp
+        TimeKeeper.cpp
+        TimeKeeper.hpp
+        timer.cpp
+        timer.h
+        wipemalloc.cpp
+        wipemalloc.h
+        )
+
+set_target_properties(projectM_static PROPERTIES
+        OUTPUT_NAME projectM
+        EXPORT_NAME static
+        )
+
+target_include_directories(projectM_static
+        PUBLIC
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer/hlslparser/src>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MilkdropPresetFactory>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/NativePresetFactory>"
+        "$<INSTALL_INTERFACE:${PROJECTM_INCLUDE_DIR}>"
+        )
+
+target_link_libraries(projectM_static
+        PUBLIC
+        OpenGL::OpenGL
+        OpenGL::GLX
+        )
+
+if(ENABLE_THREADING)
+    target_link_libraries(projectM_static
+            PUBLIC
+            Threads::Threads
+            )
+endif()
+
+add_library(projectM_shared SHARED
+        $<TARGET_OBJECTS:MilkdropPresetFactory>
+        $<TARGET_OBJECTS:NativePresetFactory>
+        $<TARGET_OBJECTS:Renderer>
+        $<TARGET_OBJECTS:hlslparser>
+        $<TARGET_OBJECTS:SOIL2>
+        $<TARGET_OBJECTS:projectM_static>
+        )
+
+set_target_properties(projectM_shared PROPERTIES
+        OUTPUT_NAME projectM
+        VERSION "${PROJECTM_LIB_VERSION}"
+        SOVERSION "${PROJECTM_SO_VERSION}"
+        EXPORT_NAME shared
+        )
+
+target_include_directories(projectM_shared
+        PUBLIC
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer/hlslparser/src>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MilkdropPresetFactory>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/NativePresetFactory>"
+        "$<INSTALL_INTERFACE:${PROJECTM_INCLUDE_DIR}>"
+        )
+
+target_link_libraries(projectM_shared
+        PUBLIC
+        OpenGL::OpenGL
+        OpenGL::GLX
```

Comment:
What's GLX for? Do we use it?

Reply:
SOIL2 automatically uses it if compiled on an X server platform, so at least on most Linux systems it's required. If I've seen it correctly, linking this target on other platforms is a simple no-op, but have to check that.

Reply:
Getting this on macOS
```
-- Configuring done
CMake Error at src/libprojectM/CMakeLists.txt:12 (add_library):
  Target "projectM_static" links to target "OpenGL::OpenGL" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/libprojectM/CMakeLists.txt:12 (add_library):
  Target "projectM_static" links to target "OpenGL::GLX" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```

Reply:
Probably a similar situation in Windows, didn't implement any specifics for those platforms yet (OpenGL in latest macOS versions is also kinda being deprecated by Apple in favor of Metal). Should compile on any recent Linux distribution though.

Reply:
Linking to `OpenGL::GL` instead and only adding `OpenGL::GLX` to the list if it's available (and in turn, SOIL2 requires it). Should resolve this configuration issue on MacOS (if the OpenGL.framework is available in the Xcode SDK).